plugins/kinetic-dp/fu-kinetic-dp-secure-aux-isp.c

```diff
@@ -0,0 +1,853 @@
+
+	*status = KT_DPCD_CMD_STS_NONE;
+
+	while (wait_time_ms > 0) {
+		if (!fu_kinetic_dp_connection_read (self, DPCD_ADDR_FLOAT_CMD_STATUS_REG,
+						     (guint8 *)&dpcd_val, 1, error))
+			return FALSE;
+
+		if (dpcd_val == KT_DPCD_CMD_STS_NONE)
+			break;	// Status is cleared by sink
+
+		if ((dpcd_val & DPCD_KT_CONFIRMATION_BIT) != DPCD_KT_CONFIRMATION_BIT) {
+			// Status is not cleared but confirmation bit is cleared, it means that target responses with failure
+			*status = dpcd_val;
+			return FALSE;
+		}
+
+		// Sleep for polling interval
+		g_usleep (((gulong) poll_interval_ms) * 1000);
+
+		if (wait_time_ms >= poll_interval_ms) {
+			wait_time_ms -= poll_interval_ms;
+		} else {
+			g_prefix_error (error, "Waiting DPCD_Isp_Sink_Status_Reg timed-out!");
+
+			return FALSE;
+		}
+	}
+
+	return TRUE;
+}
+
+// ---------------------------------------------------------------
+// In Jaguar, it takes about 1000 ms to boot up and initialize
+// ---------------------------------------------------------------
+static gboolean
+fu_kinetic_dp_secure_aux_isp_execute_isp_drv (FuKineticDpConnection *connection, GError **error)
+{
+	guint8 status;
+	guint8 read_len;
+	guint8 reply_data[6] = {0};
+
+	flash_id = 0;
+	flash_size = 0;
+	read_flash_prog_time = 10;
+
+	if (!fu_kinetic_dp_secure_aux_isp_write_kt_prop_cmd (connection, KT_DPCD_CMD_EXECUTE_RAM_CODE, error))
+		return FALSE;
+
+	if (!fu_kinetic_dp_secure_aux_isp_wait_dpcd_cmd_cleared (connection, 1500, 100, &status, error))
+	{
+		if (KT_DPCD_STS_INVALID_IMAGE == status)
+			g_prefix_error (error, "Invalid ISP driver!");
+		else
+			g_prefix_error (error, "Executing ISP driver... failed!");
+
+		return FALSE;
+	}
+
+	if (!fu_kinetic_dp_secure_aux_isp_read_param_reg (connection, &status, error))
+		return FALSE;
+
+	if (status != KT_DPCD_STS_SECURE_ENABLED && status != KT_DPCD_STS_SECURE_DISABLED) {
+		g_prefix_error (error, "Waiting for ISP driver ready... failed!");
+		return FALSE;
+	}
+
+	if (KT_DPCD_STS_SECURE_ENABLED == status) {
+		is_isp_secure_auth_mode = TRUE;
+	} else {
+		is_isp_secure_auth_mode = FALSE;
+		isp_total_data_size -= (FW_CERTIFICATE_SIZE * 2 + FW_RSA_SIGNATURE_BLOCK_SIZE * 2);
+	}
+
+	if (!fu_kinetic_dp_secure_aux_isp_read_dpcd_reply_data_reg (connection, reply_data, sizeof (reply_data), &read_len, error)) {
+		g_prefix_error (error, "Failed to read flash ID and size!");
+		return FALSE;
+	}
+
+	flash_id = (guint16)reply_data[0] << 8 | reply_data[1];
+	flash_size = (guint16)reply_data[2] << 8 | reply_data[3];
+	read_flash_prog_time = (guint16)reply_data[4] << 8 | reply_data[5];
+
+	if (read_flash_prog_time == 0)
+		read_flash_prog_time = 10;
+
+	return TRUE;
+}
+
+static gboolean
+fu_kinetic_dp_secure_aux_isp_send_isp_drv (FuKineticDpConnection *connection,
+					   gboolean is_app_mode,
+					   const guint8 *isp_drv_data,
+					   guint32 isp_drv_len,
+					   GError **error)
+{
+	g_message ("Sending ISP driver payload... started");
```

Comment:
please don't use g_message for debugging code.  I guess this was during bringup, at this point since you have nice error handling I would guess that most of these `g_message` can either go away or downgrade to `g_debug` (up to you on what's actually still useful)